docs/asciidoc/static/managing-multiline-events.asciidoc

```diff
@@ -0,0 +1,116 @@
+[[multiline]]
+=== Managing Multiline Events
+
+Several use cases generate events that span multiple lines of text. In order to correctly handle these multline events, 
+Logstash needs to know how to tell which lines are part of a single event.
+
+Multiline event processing is complex and relies on proper event ordering. The best way to guarantee ordered log 
+processing is to implement the processing as early in the pipeline as possible. The preferred tool in the Logstash 
+pipeline is the {logstash}plugins-codecs-multiline.html[multiline codec], which merges lines from a single input using 
+a simple set of rules.
+
+For more complex needs, the {logstash}plugins-filters-multiline.html[multiline filter] performs a similar task at the 
+filter stage of processing, where the Logstash instance aggregates multiple inputs.
+
+The most important aspects of configuring either multiline plugin are the following:
+
+* The `pattern` option specifies a regular expression. Lines that match the specified regular expression are considered 
+either continuations of a previous line or the start of a new multiline event. You can use 
+{logstash}plugins-filters-grok.html[grok] regular expression templates with this configuration option.
+* The `what` option takes two values: `previous` or `next`. The `previous` value specifies that lines that match the 
+value in the `pattern` option are part of the previous line. The `next` value specifies that lines that match the value 
+in the `pattern` option are part of the following line.* The `negate` option applies the multiline codec to lines that 
+_do not_ match the regular expression specified in the `pattern` option.
+
+See the full documentation for the {logstash}plugins-codecs-multiline.html[multiline codec] or the 
+{logstash}plugins-filters-multiline.html[multiline filter]plugin for more information on configuration options.
+
+==== Multiline Special Cases
+
+* The multiline codec plugin does not suport the {logstash}plugins-inputs-lumberjack[lumberjack] input plugin.
```

Comment:
@palecur based on the conversation with @allenmchan lets be more specific with what works today.

Here's what works. Can you please reword it :)

"Multiline works currently if and only if each LSF follows exactly 1 file. If LSF follows 2 or more files, the multiline codec will treat them as the same stream. This is a defect we are working to resolve soon. Until then, please be careful about using multiline codec if your environments uses these plugins"